libs/common-types/src/tokens.rs

```diff
@@ -22,61 +20,31 @@ pub enum PermissionedCurrency {
 pub enum CurrencyId {
 	// The Native token, representing AIR in Altair and CFG in Centrifuge.
 	Native,
+
 	// A Tranche token
-	Tranche(u64, [u8; 16]),
+	Tranche(PoolId, TrancheId),
 
 	/// Karura KSM
 	KSM,
 
 	/// Karura Dollar
 	KUSD,
+
 	/// Acala Dollar
 	/// Note: KUSD and AUSD will be merged into a single token, AUSD.
 	AUSD,
 
 	Permissioned(PermissionedCurrency),
-}
 
-impl TokenMetadata for CurrencyId {
-	fn name(&self) -> Vec<u8> {
-		match self {
-			CurrencyId::Native => b"Native currency".to_vec(),
-			CurrencyId::Permissioned(_) => b"Permissioned currency".to_vec(),
-			CurrencyId::Tranche(pool_id, tranche_id) => format_runtime_string!(
-				"Tranche token of pool {} and tranche {:?}",
-				pool_id,
-				tranche_id,
-			)
-			.as_ref()
-			.to_vec(),
-			CurrencyId::KUSD => b"Karura Dollar".to_vec(),
-			CurrencyId::AUSD => b"Acala Dollar".to_vec(),
-			CurrencyId::KSM => b"Kusama".to_vec(),
-		}
-	}
+	/// A foreign asset
+	ForeignAsset(ForeignAssetId),
+}
 
-	fn symbol(&self) -> Vec<u8> {
-		match self {
-			CurrencyId::Native => b"CFG".to_vec(),
-			CurrencyId::Permissioned(_) => b"PERM".to_vec(),
-			CurrencyId::Tranche(pool_id, tranche_id) => {
-				format_runtime_string!("TT:{}:{:?}", pool_id, tranche_id)
-					.as_ref()
-					.to_vec()
-			}
-			CurrencyId::KUSD => b"KUSD".to_vec(),
-			CurrencyId::AUSD => b"AUSD".to_vec(),
-			CurrencyId::KSM => b"KSM".to_vec(),
-		}
-	}
+pub type ForeignAssetId = u32;
 
-	fn decimals(&self) -> u8 {
-		match self {
-			CurrencyId::Native => 18,
-			CurrencyId::Permissioned(_) => 12,
-			CurrencyId::Tranche(_, _) => 27,
-			CurrencyId::KUSD | CurrencyId::AUSD | CurrencyId::KSM => 12,
-		}
+impl Default for CurrencyId {
+	fn default() -> Self {
+		CurrencyId::Native
```

Comment:
This is needed so that `CurrencyId` can be used as the `AssetId` type of the `AssetRegistry` pallet